src/map_extras.h

```diff
@@ -8,6 +8,7 @@
 
 #include "catacharset.h"
 #include "color.h"
+#include "generic_factory.h"
```

Comment:
This include is not necessary here, a simple forward declaration of `generic_factory` is enough for the function declaration below.